pyproject.toml

```
@@ -0,0 +1,62 @@
+[tool.poetry]
+name = "csrank"
+version = "1.2.1"
+description = "Context-sensitive ranking and choice"
+authors = ["Karlson Pfannschmidt <kiudee@mail.upb.de>"]
+license = "apache-2.0"
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Natural Language :: English",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Software Development",
+]
+repository = "https://github.com/kiudee/cs-ranking"
+documentation = "https://kiudee.github.io/cs-ranking/"
+
+[tool.poetry.dependencies]
+python = "~3.7"
```

```suggestion
python = "~3.7" # pre-compiled tensorflow 1.x is not available for newer python versions
```

It would be good to maintain the comment that explains the reason for pinning. Maybe tox is even clever enough to resolve this dependency itself, without explicit pinning?

It would be good to preserve the comment about the reason for pinning this.

---

Poetry uses SAT-solving to ensure the correct versions. So we should not need to explicitly pin this.

---

That being said, I know that it also uses this to specify the python version for PyPI. That’s why for now, let’s keep it pinned.

---

What do you mean with "for pypi"? Could you add a short comment inline that explains the reason for pinning, so that we know when we can lift the pin?